Core/Src/main.c

```diff
@@ -82,7 +96,24 @@ int main(void)
      * 7. Full regression test as much as possible. If don't know what
      * regression test is, read up on it.
      */
-
+	HAL_StatusTypeDef ret;
+	uint8_t hum_val[3]; //humidity raw value
+	uint8_t temp_val[3]; //temperature raw value
+	typedef int32_t BME280_S32_t;
+	typedef uint32_t BME280_U32_t;

```

Comment:
Why rename uint32_t?

Reply:
That's actually very common in modular libraries. It somewhat helps with portability if local type is aliased. It would allow for easier changes when code is moved to a different architecture where perhaps 32bit uint is not supported or behaves differently.